authentication/GetsWithRedirectThenPost.js

```diff
@@ -0,0 +1,157 @@
+// This script handles an authentication scheme with
+// - n GET requests giving HTTP 302 redirect while potentially collecting cookies
+// - then 1 POST request providing the login credentials to the server
+
+// Detailed usage tutorial and a php back-end can be found at https://github.com/ptrovatelli/ZAP-authentication-script-tutorial-GetsWithRedirectsThenPost
+
+// Make sure any Java classes used explicitly are imported
+var HttpRequestHeader = Java.type('org.parosproxy.paros.network.HttpRequestHeader');
+var HttpHeader = Java.type('org.parosproxy.paros.network.HttpHeader');
+var URI = Java.type('org.apache.commons.httpclient.URI');
+var AuthenticationHelper = Java.type('org.zaproxy.zap.authentication.AuthenticationHelper');
+
+
+// ------------------------------------
+// Parameters
+//   helper - a helper class providing useful methods: prepareMessage(), sendAndReceive(msg)
+//   paramsValues - the values of the parameters configured in the Session Properties - Authentication panel.
+//                      The paramsValues is a map, having as keys the parameters names (as returned by the
+//                  getRequiredParamsNames() and getOptionalParamsNames() functions below)
+//   credentials - an object containing the credentials values, as configured in the Session Properties - Users panel.
+//                      The credential values can be obtained via calls to the getParam(paramName) method. The param
+//                  names are the ones returned by the getCredentialsParamsNames() below
+function authenticate(helper, paramsValues, credentials) {
+    doLog("Authenticating via JavaScript script...");
+    var host = paramsValues.get("Hostname without trailing slash")
+    var firstGet = paramsValues.get("Fist get URI with leading slash, without trailing slash")
+
+    // GET with redirects
+    var msg = doGet(host + firstGet, helper);
+    var statusCode = msg.getResponseHeader().getStatusCode();
+    while (statusCode == 302) {
+        // Add the request/response to ZAP history tab
+        AuthenticationHelper.addAuthMessageToHistory(msg);
```

Comment:
Is there a reason this doesn't use `helper`?

Reply:
This one is static, using `helper` would lead to an error.

Reply:
Ok